### PR TITLE
Add site configs for Narwhal, Casper, Ubuntu 22.04, update documentation for Python 3.10, add OpenMP variants for eckit, fckit, ecmwf-atlas

### LINKED ISSUE #331: Port spack-stack to Narwhal
**Is your feature request related to a problem? Please describe.**
Port spack-stack to NRL's Narwhal system.

**Additional context**
n/a

### LINKED ISSUE #381: Port spack-stack to Casper
**Is your feature request related to a problem? Please describe.**
Support for Casper is needed for the JCSDA data ingest tasks.

**Describe the solution you'd like**
The stack is working on Cheyenne, it should be relatively easy to get it over to Casper.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Add site configs for Narwhal, Casper, Ubuntu 22.04, update documentation for Python 3.10, add OpenMP variants for eckit, fckit, ecmwf-atlas.

Notes:
- Narwhal is using the existing `cray-python` module, this is now possible (see #394)
- Narwhal uses a source cache (spack mirror) similar to Hera, because not all packages are accessible from the system itself
- Casper is using a new way to work around the `lmod` priority settings on Cheyenne/Casper (thanks to help from UCAR IT)
- Ubuntu 22.04 is using Python 3.10 - all previous build errors where addressed in the authoritative spack repo (and brought over) or in our fork
- Additional updates to the documentation where made regarding the use of Python 3.10

### Issues

Fixes #381
Fixes #331 

### Dependencies

Waiting on https://github.com/NOAA-EMC/spack/pull/189

### Testing

- Tested on Casper, Narwhal, Ubuntu 22.04 (AWS EC2)
- **CI tests are still running**

### CODE REVIEW

**Review by AlexanderRichert-NOAA — APPROVED** [review]

File: .gitmodules
Comment (ulmononian) [inline .gitmodules]: are the url/branch in .gitmodules intentionally changed here or was this only done for testing? i see this PR is waiting on https://github.com/NOAA-EMC/spack/pull/189, so it would make sense.

**Review by ulmononian — APPROVED** [review]
i do not have access to narwhal or casper, but the changes/additions and updates to the docs look solid to me. i'll defer to others & the CI checks for testing.

File: .gitmodules
Comment (climbfuji) [reply .gitmodules]: I see you passed the "careful review" test ;-) Everyone asks the question the first time, rightfully, because this is not something that should get merged. It's needed - like for the ufs-weather-model - so that the CI tests check out the updated spack code and test with it. Once the spack PR is merged (I will wait for approvals and the CI tests), then `.gitmodules` gets reverted and the spack submodule pointer will be pointed back to the (new) head of jcsda_emc_spack_stack branch in the noaa-emc spack fork. Only after that this PR can be merged.

Comment (ulmononian) [reply .gitmodules]: @climbfuji haha, glad i was able to catch a red herring here. what was the old saying? teach a man to fish and he might leave extraneous comments on github PR reviews for life? 

but that now makes sense and i appreciate the description. noted for future PRs.

**Comment by climbfuji** [comment]
The macOS CI test failures are new ... and not related to this PR. They seem to have to do with an update of the macOS images used by Github actions. I'll try to fix them nonetheless in this PR, hopefully.

**Comment by climbfuji** [comment]
Finally got the CI tests fixed on macOS!

**Comment by climbfuji** [comment]
> Finally got the CI tests fixed on macOS!

The Intel CI tests timed out for the skylab-dev environment a few packages before completing. But these tests finished successfully earlier in this PR, so that is ok. Merging now.

**Pull request merged**
